node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -180,10 +180,17 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     if (blockHashes.isEmpty) {
       Future.unit
     } else {
-      val peerMsgSenderF = peerManager.randomPeerMsgSenderWithService(
-        ServiceIdentifier.NODE_NETWORK)
-      peerMsgSenderF.flatMap(
-        _.sendGetDataMessage(TypeIdentifier.MsgWitnessBlock, blockHashes: _*))
+      val syncPeerOpt = getDataMessageHandler.syncPeer
+      syncPeerOpt match {
+        case Some(peer) =>
+          peerManager
+            .peerData(peer)
+            .peerMessageSender
+            .sendGetDataMessage(TypeIdentifier.MsgWitnessBlock, blockHashes: _*)
```

Comment:
I think it still makes sense to call from a random peer here

Reply:
At the moment, we simply ignore any messages from a peer that's not `syncPeer`. What you say is the ultimately the goal though.